dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
+                )
+                .pieces[0]
+                .get_metadata()
+            )
+        dataset = pq.ParquetDataset(
+            paths, filesystem=fs, filters=filters, **dataset_kwargs
+        )
+        if proxy_metadata:
+            dataset.metadata = proxy_metadata
+    elif fs.isdir(paths[0]):
+        # This is a directory.  We can let pyarrow do its thing.
+        allpaths = fs.glob(paths[0] + fs.sep + "*")
+        base, fns = _analyze_paths(allpaths, fs)
+        if "_metadata" in fns and "validate_schema" not in dataset_kwargs:
+            dataset_kwargs["validate_schema"] = False
+        dataset = pq.ParquetDataset(
+            paths[0], filesystem=fs, filters=filters, **dataset_kwargs
+        )
+    else:
+        # This is a single file.  No danger in gathering statistics
+        # and/or splitting row-groups without a "_metadata" file
+        base = paths[0]
+        fns = [None]
+        dataset = pq.ParquetDataset(
+            paths[0], filesystem=fs, filters=filters, **dataset_kwargs
+        )
+
+    return dataset, base, fns
+
+
+def _gather_metadata(
+    paths, fs, split_row_groups, gather_statistics, filters, dataset_kwargs
+):
+    """ Gather parquet metadata into a single data structure.
+
+        Use _metadata or aggregate footer metadata into a single
+        object.  Also, collect other information necessary for
+        parquet-to-ddf mapping (e.g. schema, partition_info).
+    """
+
+    # Step 1: Create a ParquetDataset object
+    dataset, base, fns = _get_dataset_object(paths, fs, filters, dataset_kwargs)
+    if fns == [None]:
+        # This is a single file. No danger in gathering statistics
+        # and/or splitting row-groups without a "_metadata" file
+        if gather_statistics is None:
+            gather_statistics = True
+        if split_row_groups is None:
+            split_row_groups = True
+
+    # Step 2: Construct necessary (parquet) partitioning information
+    partition_info = {"partitions": None, "partition_keys": {}, "partition_names": []}
+    fn_partitioned = False
+    if dataset.partitions is not None:
+        fn_partitioned = True
+        partition_info["partition_names"] = [
+            n for n in dataset.partitions.partition_names if n is not None
+        ]
+        partition_info["partitions"] = dataset.partitions
+        for piece in dataset.pieces:
+            partition_info["partition_keys"][piece.path] = piece.partition_keys
+
+    # Step 3: Construct a single `metadata` object. We can
+    #         directly use dataset.metadata if it is available.
+    #         Otherwise, if `gather_statistics` or `split_row_groups`,
+    #         we need to gether the footer metadata manually
+    metadata = None
+    if dataset.metadata:
+        # We have a _metadata file.
+        # PyArrow already did the work for us
+        schema = dataset.metadata.schema.to_arrow_schema()
+        if gather_statistics is None:
+            gather_statistics = True
```

Comment:
Here and for the single-file case, does this still imply having to parse the values, possibly unnecessarily?

Reply:
No - We still make sure that there are filters and/or an index before actually gathering metadata.  The check here is just to establish we "will" do it if this criteria is met.